flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherOperationCaches.java

```diff
@@ -40,13 +42,19 @@ public DispatcherOperationCaches() {
     @VisibleForTesting
     public DispatcherOperationCaches(Duration cacheDuration) {
         savepointTriggerCache = new CompletedOperationCache<>(cacheDuration);
+        checkpointTriggerCache = new CompletedOperationCache<>(cacheDuration);
     }
 
     public CompletedOperationCache<AsynchronousJobOperationKey, String> getSavepointTriggerCache() {
         return savepointTriggerCache;
     }
 
+    public CompletedOperationCache<AsynchronousJobOperationKey, Long> getCheckpointTriggerCache() {
+        return checkpointTriggerCache;
+    }
+
     public CompletableFuture<Void> shutdownCaches() {
-        return savepointTriggerCache.closeAsync();
+        return CompletableFuture.allOf(
```

Comment:
FutureUtils#completeAll is preferably because it doesn't drop exceptions should multiple futures fail.